internal/app/storage/file/file.go

```diff
@@ -0,0 +1,105 @@
+package file
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+)
+
+// Record представляет запись для хранения URL.
+type Record struct {
+	UUID        string `json:"uuid"`
+	ShortURL    string `json:"short_url"`
+	OriginalURL string `json:"original_url"`
+}
+
+// FileStore – реализация URLStore, использующая файл для персистентности.
+type FileStore struct {
+	filePath string
+	data     map[string]string
+}
+
+func NewFileStore(filePath string, sugar *zap.SugaredLogger) (*FileStore, error) {
+	store, err := loadStorageFromFile(filePath, sugar)
+	if err != nil {
+		return nil, err
+	}
+	return &FileStore{filePath: filePath, data: store}, nil
+}
+
+func (fs *FileStore) Save(shortURL, originalURL string) (string, error) {
+	fs.data[shortURL] = originalURL
+	rec := Record{
+		UUID:        "", // для FileStore не требуется генерация UUID
+		ShortURL:    shortURL,
+		OriginalURL: originalURL,
+	}
+	return shortURL, appendRecordToFile(rec, fs.filePath)
```

Comment:
[SUGGESTION] This is a completely incorrect approach. We should save and load data to the file system only once. The Memento pattern should have been implemented here.